src/extractor/class-feed-extractor.php

```diff
@@ -0,0 +1,105 @@
+<?php
+/**
+ * Feed_Extraction class file
+ *
+ * @package feed-consumer
+ */
+
+namespace Feed_Consumer\Extractor;
+
+use Feed_Consumer\Contracts\Processor;
+use Feed_Consumer\Contracts\With_Settings;
+use Mantle\Http_Client\Pending_Request;
+use Mantle\Http_Client\Response;
+use RuntimeException;
+
+/**
+ * Feed Extractor
+ *
+ * Used to fetch common feeds and extract the data. Supports basic HTTP
+ * authentication.
+ */
+class Feed_Extractor extends Extractor implements With_Settings {
+	/**
+	 * Setting for the feed URL.
+	 *
+	 * @var string
+	 */
+	public const SETTING_FEED_URL = 'feed_url';
+
+	/**
+	 * Setting for the username.
+	 *
+	 * @var string
+	 */
+	public const SETTING_USERNAME = 'feed_username';
+
+	/**
+	 * Setting for the password.
+	 *
+	 * @var string
+	 */
+	public const SETTING_PASSWORD = 'feed_password';
+
+	/**
+	 * Response for the feed.
+	 *
+	 * @var \Mantle\Http_Client\Response
+	 */
+	protected $response;
+
+	/**
+	 * Settings to register.
+	 */
+	public function settings(): array {
+		return [
+			static::SETTING_FEED_URL => new \Fieldmanager_TextField( __( 'Feed URL', 'feed-consumer' ) ),
+			static::SETTING_USERNAME => new \Fieldmanager_TextField( __( 'Username (optional)', 'feed-consumer' ) ),
+			static::SETTING_PASSWORD => new \Fieldmanager_TextField( __( 'Password (optional)', 'feed-consumer' ) ),
```

Comment:
It looks like Fieldmanager is a dependency, but I don't see where we're asserting that it exists in this plugin. Just something to note.

Reply:
Included an admin notice in a followup